beacon-chain/blockchain/forkchoice_update_execution.go

```diff
@@ -103,7 +103,7 @@ func (s *Service) forkchoiceUpdateWithExecution(ctx context.Context, args *fcuCo
 	}
 
 	// Only need to prune attestations from pool if the head has changed.
-	if err := s.pruneAttsFromPool(args.headBlock); err != nil {
+	if err := s.pruneAttsFromPool(s.ctx, args.headState, args.headBlock); err != nil {
```

Comment:
Note this is using `headState`

Reply:
See https://github.com/prysmaticlabs/prysm/blob/56208aa84d55e502a4746383148cd4654b6c9108/beacon-chain/blockchain/process_block_helpers.go#L56. `fcuArgs.headState` is what's passed to the pruning function, so it's actually the post state of the incoming block.